fpnge_main.cc

```diff
@@ -22,15 +22,44 @@
 #include "fpnge.h"
 #include "lodepng.h"
 
+static int print_usage(const char *app) {
+  fprintf(stderr, "Usage: %s [options] in.png out.png\n", app);
+  fprintf(stderr, "  -1..%d  compression level (default %d)\n",
+          FPNGE_COMPRESS_LEVEL_BEST, FPNGE_COMPRESS_LEVEL_DEFAULT);
+  fprintf(stderr, "  -r<n>  run <n> repetitions and report\n");
+  return 1;
+}
+
 int main(int argc, char **argv) {
   if (argc < 3) {
-    fprintf(stderr, "Usage: %s in.png out.png [num_reps]\n", argv[0]);
-    return 1;
+    return print_usage(argv[0]);
+  }
+
+  int comp_level = FPNGE_COMPRESS_LEVEL_DEFAULT;
+  size_t num_reps = 0;
+
+  int arg_p = 1;
+  for (; arg_p < argc; arg_p++) {
+    if (argv[arg_p][0] != '-')
+      break;
+    char opt = argv[arg_p][1];
+    if (opt >= '1' && opt <= '0' + FPNGE_COMPRESS_LEVEL_BEST) {
```

Comment:
We should probably also support passing `--` to separate options from files (in the rare case someone has files starting with -)

Reply:
Most people just do `./-file` for those cases. You're free to add `--` support if you wish though.

Reply:
Fair point.